### ISSUE TITLE
Fix false positives for level 4 math functions in length-zero-no-unit

### ISSUE THREAD

**Issue by tomyo** [stylelint/stylelint#5199]
I have the following css, and if I take out the unit from 0rem, as stylelint suggests, it breaks.

What I have:

```css
right: calc(-1 * max(100vw, 0rem));
```
What stylelint suggest me to do: (length-zero-no-unit error)

```css
right: calc(-1 * max(100vw, 0));
```

The later option without unit is not a valid css line in my browser.

What to expect:
length-zero-no-unit should be omitted inside max/min functions.

Reference: https://stackoverflow.com/questions/62520998/why-doesnt-min-or-max-work-with-unitless-0

**Comment by jeddy3** [comment]
Thanks for the report.

I can reproduce this using [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYKBOBLA5gCwBd4UwkAbMACgFoBGFAKhQFskAPS2gBi4DcB3ADQouGGMwCUEgNwoAvgB0AdiEEgAzgE8lBdgnDr1q8BCUAzXPuDKUKBSAwBXMjHX3i1pbdv2kBak4u1EoQ1I5KANYh-EruKAROMII23vYARmQQYBHBoeIADgSacQmOSSk+JpkYudRYSrzkWFDUeDBsJYnJXqkmzMwwOrUFRZ1l3d52ILBgZEgYflim1BlZOSHUUI75ZFikBDDUYI7qBBDM1PkYEPkwGARYrmPlPZUzcwsPy6vZtVs7ez8hyuNzuDyeCBQAG0Kt5Si9JrZPIjergQmI4jDXiippAlOoYMcHrxDv9dvtXNR+FgCHhNlgzGY7oN-I0yGU3CBYZMALrc+SwnkTXrveaLb6ZX4bdR4CD3PBIJQtEG3e6aagQEkYbCwTnEeHCypmUz+MxIZhYMjqjZkwEHYLmiH6rqw+zGoZmi1W2oWwz1HDUHCDO57aiey3qiIwTT8OVQZ6GqZmcJgL5KI7kMC1eps5oJ10gZNKVNLdOkCi1cLqfJIMAwFqgz5y-PYt0ptPUXZKGDzQMLKCPIYbEL43RK+YtAdiEumFuTexRzRmBYDTaEj7i9MbLDMfJyscEOe9AYDpBhnsERxiB2rjbhKIQGJHypKR0tHA66jzHvqbMNJrxpCBoFjarh1kq-rUNWhIMnsNLFEBLqtiANrbOSQJfv4O57vcATOE68RIfOKGhLaFJQTALipnKeqEeMIF5LuRRQRAV51s+UwbO0CRngSFqQJk+IcfY27-rsk6sRkhzqHMMpHOcAw6LRwHISqYLWqE97RLEiH0chBJUWcNT5ASjhQKEsxIIYlaRNpwkaJRhJGZcpnmdQjmKf4d62Y+OnOnpxEGU5crUEUtw2Q+T66QivSnNgSgBsOMD8F2MD2eENIRXZ0WJr4-iBIcIzqqlKwwMaGKQliKK+GQ-BIJonKJki-KVO0daFJiLW9D8EQuNZSBmAcNTqI6N6HD1fWNV1RpYBgpzBK4ByAV1QpdfYaJymllXTVMA1DfJ-QsvYK38oo2Krchfh4UEr6rqQBJxPYmT8Hc9i5cg+X4WNUE1nWX6Da9kI1XVDVQf6QSpW9BZXQVFH8RAgkLSl9SHHtgPEMD9VTchPVHJk6iQWkCx-UVnYo6V5VbRjKEwFqUM45KOSzBABMJSsxOHN2yPdv9Q2PcgtVY-TxG48zrMBkTtac8lJVpGVm38+QIO-swzgPGT3bC91jN4yzhMcz9UsUwrQMC8rYMJRDKNa5UuOgko+tG1zJVoxgiuC6DqtkOrkOqAWdu3A7bOS391ZG677vm+LVua37DNrBqgeO6Hv3jfLFXU0rQtx8RAkhW0bAZg9pvPYD715zUBedoMOC0vzMr7jbuIKSy7lMcV5Ny5TnXYpUWcNU3zW970bUwB123D8RFhzV5i31sdk8oBdOJTBtGfQjt9inJoLhdv4kD9IqUCcidw9nby5ct0M-B4DSrip3U+LNFTu0e9jucnGcFxqWqbeFB3PMu4m2IFVREmMB450RMiHE9hR7jxAZvZAAMajHFOOcS41xVSjEgSvN0s15rdlOPPLki9l4wJAGvF+oDcFIP2gfTyg8wEUKfrAC2OAY7jUZgvFefIz6CneqKT4pYViKgDGHP6EdTbdjpjgqYgjNwiLZuItO3dTb93fiKdcYoOy4z4nsBGyxnbk0kZnN+1AvY+2trI+w8idE6z0XndMyjebo1flHcGhxfYCK0UIiUCcHEGKcQ-IB68RK0zLgWWxwjdEeI1ocVghcommForQbxllfHplxjxS0kEAmCUjtnNJG4OyOKRi7ZBBTPZqywHEpuNifEKNKc4kxbisZsI4XU6YDSSmBMNn9EJL8wkyKKdo4RpNZbpyofycBGjJjQOqiAOBh4J4rz7sgtc6TNyMKnvg2eRDlqkKaqiHA6IpmLzWXQq+yyjnEXqATVh0dPGdy4SQnhp1+EFiLDOMsCkzxGJ5i0mZ5jqm1OsYWdswj6G8QfoCs2bTHmgvel8nph8+kqOAZxcJbswXIuEQkv+zFUq0S4EiiFhjRr3UGSAUu2LSXFg7DWMQOg2gEl-P8w4dzn6VJViCrxnyyXpkZSyFllJnGctgPzaRdx2lPNjnS75VJb4HGaRUtRb9On1FgDoLZkIABM70XAJVpNQAAXncUI3kaT2XxeMlGKT3oniwGeJkfgryHCaTC1VpjladMdc6i8br5KI2cQMyVWLfX1ideeV115bruqslSmlEbTzRsvNeIVzLXCiofuKqlUraUFj9amwNCwEqHEbH4EKKq+Zqp9WCotLq02HFLUGROdxK3DWCZM7lyao0AEcygYA7vNKFZTjFetaVU72NS+XISLQOu4w796-LRS47F3r4WxNncRedg6l0HVYKu0NUjw1gq4gjRVd9lH2Q2L6cW7klQajMCxNi0tuYvxUsRJQjhmByxqC4JAA42ZmuuD2s9P6-21GyV2AMIGWb2R-sxSl-Mk1gthrawB3aVlMPUcC6diL+TzKYUsnuqyky7IWvs7hOIyELMoaR1Zvh1n0KOq8mj7zzrvSCtRGofgEhYDSI4e0IcowEF-GKlhebT1cccjxzC-HBP2grc5atrihkRP0rJ5yfHsCKfLaqDtR6sPU3zZ07jzkD5pHqIZ1T67J2zKmOZkKlnrMqa7aojdECZOGRCrqcCUBFTLt-a5kKw5ljXpygWJz-6sAjpXeytdYHvPBRi3F1FIbjOYuGVFrTIUbXtzicS5LcmTIwDMhZWSv5kMl0fBpwKuXjKuQq1ZX8GbaRZvEzmyTYbsuaZ841srbkPKt2q9TVDxXnKlfK+5Fwnkg2GIgLoNM-NzKOCkmZhroVNDhVG1McbBYMr7wTSh2rBbkJsjKNQRcsYMAtF209U7nSLtPLS4ehLsLcMWJnVY96z2yaveheHCdQKEXbt6H93YAOjMeay3V8H5BLuQ-8Pl-+hW4hcBSGdOQIA5BAA).

Fortunately, the rule is already set up to ignore _math functions_. However, we only have `calc` listed in [our reference (`lib/reference/mathFunctions.js`)](https://github.com/stylelint/stylelint/blob/master/lib/reference/mathFunctions.js). We'll need to add all the [level 4 math functions](https://drafts.csswg.org/css-values-4/#math-function) as they all have their own special rules.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide. Adding a single test for `min` should suffice.
